contracts/dao/neutron-chain-manager/src/state.rs

```diff
@@ -0,0 +1,9 @@
+use crate::msg::Strategy;
+use cosmwasm_std::Addr;
+use cw_storage_plus::Map;
+
+/// Defines a mapping from an address to a strategy associated with the address.
+pub const STRATEGIES_ALLOW_ALL: Map<Addr, Strategy> =
+    Map::new("chain-manager-strategies-allow-all");
+pub const STRATEGIES_ALLOW_ONLY: Map<Addr, Strategy> =
+    Map::new("chain-manager-strategies-allow-only");
```

Comment:
Maybe I don't see something, but:
How about redoing the types to be something like:

```rust
STRATEGIES: Map<Addr, Strategy>

enum Strategy {
  AllowAll,
  AllowOnly(Vec<Permission>)
}

enum Permission {
  //... same as before
}

```
It simplifies code a lot, removes checks for addr being only either AllowAll or AllowOnly

Reply:
+

Reply:
done. please check!